src/Host/Monitor/Impl/MainWindow.xaml.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Windows;
+
+namespace Microsoft.R.Host.Monitor {
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window {
+        public static MainWindow _currentWindow;
+        public MainWindow() {
+            InitializeComponent();
+            Loaded += MainWindow_Loaded;
+            _currentWindow = this;
+            Closing += MainWindow_Closing;
+        }
+
+        private async void MainWindow_Closing(object sender, System.ComponentModel.CancelEventArgs e) {
```

Comment:
Same thing for all the `async void` methods here - what happens if any of the code inside throws?